halo2_backend/src/plonk/permutation.rs

```diff
@@ -17,12 +17,13 @@ pub mod verifier;
 
 /// The verifying key for a single permutation argument.
 #[derive(Clone, Debug)]
-pub struct VerifyingKey<C: CurveAffine> {
+pub(crate) struct VerifyingKey<C: CurveAffine> {
     commitments: Vec<C>,
 }
 
 impl<C: CurveAffine> VerifyingKey<C> {
     /// Returns commitments of sigma polynomials
+    #[allow(unused)]
```

Comment:
Since this is `pub` and not `pub(crate)` I think it's best to remove the `#[allow(unused)]`.  This way it will stay like before the frontend-backend split: 
https://github.com/privacy-scaling-explorations/halo2/blob/73408a140737d8336490452193b21f5a7a94e7de/halo2_proofs/src/plonk/permutation.rs#L91-L92

Reply:
How about removing these `unused` methods, not just `#[allow(unused)]` attr?
I think they are not used anywhere.

Reply:
AFAIK, these methods were not used elsewhere before frontend-backend split.
If we wanna keep the methods as is, the CI complains. (I need to add the `#[allow(unused)]` attr)

Reply:
> AFAIK, these methods were not used elsewhere before frontend-backend split. If we wanna keep the methods as is, the CI complains. (Need to add the `#[allow(unused)]` attr)

I think you mean these methods were not used anywhere inside this repository right?  But they were publicly exposed, so external projects could have been using this method; that's why I propose keeping it `pub`.  If they are `pub` and without the `#[allow(unused)]`, does the CI complain?

Reply:
Yes, the CI complains. 
Like,
```
warning: associated function `with` is never used
...
= note: `#[warn(dead_code)]` on by default
```